src/main/java/org/folio/utils/UserPermissionsUtil.java

```diff
@@ -63,12 +67,31 @@ public static void verifyUserHasManagePermission(List<String> newAcqUnitIds, Lis
     }
   }
 
+ public static void  verifyUserHasInvoicePayPermission  (Invoice.Status newInvoiceStatus, Invoice.Status statusFromStorage,   Map<String, String> okapiHeaders ){
+
+    if (isInvoiceStatusUpdated(newInvoiceStatus, statusFromStorage)){
+      if(!(newInvoiceStatus.toString().equals("Approved")) &&!(newInvoiceStatus.toString().equals("Paid"))&& isUserDoesNotHaveDesiredPermission(PAY, okapiHeaders)){
```

Comment:
don't need to convert to strings, you can just compare enums with ==

Reply:
formatting is incorrect in this line, see how another if statement written and use the same formatting

Reply:
we should not check both permissions in this method.
We should have 2 separate methods, in one of them check Approve permission, in another - Pay.
These permissions are not connected between each other